app/controllers/comments_controller.rb

```diff
@@ -1,32 +1,31 @@
 class CommentsController < ApplicationController
-  before_action :find_article, only: %i[create]
   before_action :find_comment, only: %i[destroy]
-
   http_basic_authenticate_with name: 'user', password: 'secret', only: :destroy
 
   def create
-    @comment = @article.comments.create(comments_params)
-    redirect_to article_path(@article)
+    @comment = Comment.new(comments_params)
+    @comment.article_id = params[:article_id]
+    begin
+      @comment.save!
+    rescue ActiveRecord::RecordInvalid
+    end
+    redirect_back fallback_location: articles_url
   end
 
   def destroy
-    if @comment.article_id == params[:article_id].to_i
-      @comment.destroy
-    end
+    @comment.destroy if @comment.article_id.eql? params[:article_id].to_i
     redirect_back fallback_location: articles_url
   end
-  
+
   private
 
   def comments_params
-    params.require(:comment).permit(:commenter, :body, :status)
-  end
-
-  def find_article
-    @article = Article.find(params[:article_id])
+    params.require(:comment).permit(:commenter, :body, :status, :article_id)
```

Comment:
`article_id` is not part of the comment body.

Reply:
Done !